hs/WebParsing/TimeTableParser.hs

```diff
@@ -17,15 +18,111 @@ import WebParsing.TimeConverter
+                "and",
+                "-",
+                "[",
+                "]",
+                ""]
+
+-- | Parses the timetable main page, retrieves the tag text and gets the department
+-- codes in a list of lists of T.Text.
+getDeptCode :: [Tag String] -> [[T.Text]]
+getDeptCode tags = 
+    let tagsText = map fromTagText $ filter isTagText tags
+        filterDeptInfo = map (filterList bracketLeft) tagsText
+        removedEmptyStr = filter (\x -> x /= "") filterDeptInfo
+        removedChars = map (removeUnwantedChar bracketLeft) removedEmptyStr
+        listOfWords = map words removedChars
+        relevantWords = map (filter (`notElem` notDeptCodes)) listOfWords
+        deptCodesStr = map (filter (\x -> isUpperString x /= False)) relevantWords
+        deptCodesText = map (map T.pack) deptCodesStr
+    in deptCodesText
+
+-- | Parses the timetable main page, retrieves the tag text and gets the department
+-- names in a list of T.Text.
+getDeptName :: [Tag String] -> [T.Text]
+getDeptName tags = 
+    let tagsText = map fromTagText $ filter isTagText tags
+        containsBrackLeft = filter (\x -> containsChar bracketLeft x == True) tagsText
+        convertToStr = map T.pack containsBrackLeft
+        deptNamesText = map head $ map (T.splitOn (T.pack bracketLeftString)) convertToStr
+    in deptNamesText
+
+-- | Takes a Char and a String and returns whether or not that Char is in the String.
+containsChar :: Char -> String -> Bool
+containsChar _ [] = False
+containsChar y (x:xs)
+    | y == x = True
+    | otherwise = containsChar y xs  
+
+-- | Takes a Char and a String and checks whether that char is in the String. If it
+-- is, it returns the original string, otherwise it returns an empty string.
+filterList :: Char -> String -> String 
+filterList _ [] = ""
+filterList y (x:xs) 
+    | y == x = (t:ts)
+    | otherwise = filterList y xs  
+    where (t:ts) = (x:xs)
+
+-- | Removes the chars before and including the Char input in the given String.
+removeUnwantedChar :: Char -> String -> String
+removeUnwantedChar _ [] = ""
+removeUnwantedChar y (x:xs)
+    | y == x = xs
+    | otherwise = (x:xs)
+
+-- | Returns whether all the elements of the String are uppercase.
+isUpperString :: String -> Bool
+isUpperString [] = True
+isUpperString (x:xs)
+    | Data.Char.isUpper x == True && isUpperString xs == True = True
+    | otherwise = False
+
+-- | Constructs an association list given a list of T.Text and a list of lists of T.Text.
+constructAssociationList :: [T.Text] -> [[T.Text]] -> [([T.Text], T.Text)]
+constructAssociationList _ [] = []
+constructAssociationList [] _ = []
+constructAssociationList (x:xs) (y:ys) = [(y, x)] ++ constructAssociationList xs ys
+
+-- | Inserts the Department association list in the database.
+insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m (Key Department)
+insertDepartment (code, name) = 
+    Database.Persist.Sqlite.insert (Department code name)
```

Comment:
insert_ is better suited in this case. Thanks again, I'm new to Haskell and the different libraries but your comments are really helpful!